app/models/update_log.rb

```diff
@@ -1,17 +1,51 @@
+# frozen_string_literal: true
+
 class UpdateLog
+  ####################
+  # CONSTANTS        #
+  ####################
+  # We want to keep a maximum number of 10 updates, so the last key we want to save is the ninth
+  MAX_UPDATES_KEY = 9
 
-  def self.setting_record
-    Setting.find_or_create_by(key: 'metrics_update')
+  def log_updates(times)
+    add_new_log(times)
+    delete_old_log
+    log_delay
+    save_setting
   end
 
-  def self.log_update(time)
-    setting = self.setting_record
-    setting.value["last_update"] = time
-    setting.save
+  def updates
+    return unless setting_record.value['constant_update']
+    { "last_update" => setting_record.value['constant_update'].values.last['end_time'] , "average_delay" => setting_record.value['average_delay']}
   end
 
-  def self.last_update
-    self.setting_record
-    Setting.find_by(key: 'metrics_update').value['last_update']
+  def setting_record
+    @setting ||= Setting.find_or_create_by(key: 'metrics_update')
   end
-end
\ No newline at end of file
+
+  private
+    def add_new_log(times)
+      setting_record.value['constant_update'] ||= {}
+      @last_update = setting_record.value['constant_update'].keys.last
+      this_update = @last_update ? @last_update + 1 : 0
+
+      setting_record.value['constant_update'][this_update] = times
+    end
+
+    def delete_old_log
+      return unless @last_update&. >= MAX_UPDATES_KEY
+      setting_record.value['constant_update'].delete(@last_update - MAX_UPDATES_KEY)
+    end
+
+    def log_delay
+      constant_update = @setting.value['constant_update']
+      return unless constant_update&.length&. > 1
+      times = constant_update.values.map { |log| log['end_time'] }
+      seconds = (times[times.length - 1] - times[0]) / (times.length - 1)
+      @setting.value['average_delay'] = seconds.to_i
```

Comment:
The subtraction of times is where I get the weird results locally; using `to_f` on datetimes will convert it into seconds consistently.

We can also use some of Array's methods to make this read more clearly:

```ruby
      average_time = (times.last.to_f - times.first.to_f) / (times.count - 1)
      @setting.value['average_delay'] = average_time
```